### PR TITLE
[native]Add native dynamic filter stats to operator stats for hbo

### PR DESCRIPTION
**Pull request by xiaoxmeng**
Exports the dynamic filter stats from native worker to coordinator:
1. add DynamicFilterStats in OperatorStats and update presto protocol. 
2. Presto task collects the dynamic filter stats from velox.

### CODE REVIEW

**Comment by steveburnett** [comment]
I don't see documentation for me to review, perhaps I don't need to review this. Let me know if there's something I can help with! The description is empty, does this PR need a release note entry?

**Comment by xiaoxmeng** [comment]
> I don't see documentation for me to review, perhaps I don't need to review this. Let me know if there's something I can help with! The description is empty, does this PR need a release note entry?

@steveburnett Hi, the pr description is updated.

File: presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Records the dynamic filter execution stats for an operator on native worker.
+ */
+@ThriftStruct
+public class DynamicFilterStats
+{
+    private final Set<PlanNodeId> producerNodeIds;
```
Comment (feilong-liu) [inline presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java:33]: These PlanNodeIds will be the PlanNodeId of the joins which has dynamic filter pushdown?
How do we know which nodes these filters are pushed down to?

Comment (xiaoxmeng) [reply presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java:33]: DynamicFilterStats is set in operator stats of those nodes that has dynamic filters.

Comment (feilong-liu) [reply presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java:33]: For example, if we have a query plan like:
join <- (probe side) project <- filter <- scan
where the dynamic filter pushdown is enabled, we should have project, filter and scan all in the producerNodeIds, right?

Comment (feilong-liu) [reply presto-main/src/main/java/com/facebook/presto/operator/DynamicFilterStats.java:33]: Synced offline, the plan node ids will be the where the dynamic filter is from, which is the join node in the example I showed above.

**Review by feilong-liu — DISMISSED** [review]
Thanks for the change!

File: presto-native-execution/presto_cpp/main/PrestoTask.cpp [hunk 2]
```diff
@@ -552,6 +552,17 @@ void PrestoTask::updateMemoryInfoLocked(
...
+namespace {
+presto::protocol::DynamicFilterStats toPrestoDynamicFilterStats(
+    const velox::exec::OperatorStats& veloxOpStats) {
+  presto::protocol::DynamicFilterStats dynamicFilterStats;
+  for (const auto& nodeId : veloxOpStats.dynamicFilterStats.producerNodeIds) {
+    dynamicFilterStats.producerNodeIds.emplace_back(nodeId);
+  }
+  return dynamicFilterStats;
+}
+} // namespace
```
Comment (tanjialiang) [inline presto-native-execution/presto_cpp/main/PrestoTask.cpp:564]: Move to merge with top anonymous namespace for all helpers?

**Review by tanjialiang — APPROVED** [review]
Thanks %minor

**Review by feilong-liu — APPROVED** [review]

**Review by tanjialiang — APPROVED** [review]

**Pull request merged**